src/BaseRepository.php

```diff
@@ -136,11 +136,13 @@ public function insert(array $fields)
 	{
 		$return = $this->dba->insert(static::$table_name, $fields);
 
-		if ($return) {
-			$fields['id'] = $this->dba->lastInsertId();
-			$return = $this->create($fields);
+		if (!$return) {
+			throw new HTTPException\InternalServerErrorException('Unable to insert new row in table "' . static::$table_name . '"');
 		}
 
+		$fields['id'] = $this->dba->lastInsertId();
```

Comment:
I'm not a db-guy, so @annando is it possible to have a second instance/thread and a race condition with it? Wouldn't it be better to have a `$this->dba->transaction();` `$this->dba->commit();` around it?

Reply:
`lastInsertId` is valid in the same context. We should only ensure to not encapsulate the previous `insert` too much. Means: when the `insert` command would call some other database `insert` afterwards, then we had a problem. If this isn't the case, then it's totally safe.